### DISCUSSION TITLE
1.2 release status

### DISCUSSION THREAD

**Discussion by dburgener** [TresysTechnology/selint#185] [General]
My target was to make the 1.2.0 release on 12/15.  It's not looking to me like #183 will be in in time for that (I'd really like it in by now to feel comfortable with a release including it tomorrow so there's some time for everyone to take it for a spin).

In my opinion, it makes sense to wait until tomorrow AM and see what kind of shape #183 is in at that point.  If it's close, we can keep waiting on it.  If not, we can decide to go ahead without it for now and get it in 1.3.0.

I'm not tracking anything else we're looking to get in except the CHANGELOG updates and any last minute bug fixes that come up.

Once either #183 is merged, or we decided to not go with it, I'll spend a while playing with the latest version and see if I can get anything to break.  I'd appreciate it if anyone else with time could do the same.

My preference is to release before it gets too late in the week in order to give time for a 1.2.1 before the holidays if needed.

**Comment by dburgener** [comment]
@gdesrosi has indicated to me offline that he doesn't expect to have #183 ready for this release, and I agree based on the current state.  So the plan will be to go ahead with the features we have now plus any fixes found in the next day or so.  I'll plan on cutting a release on Wednesday afternoon unless anyone objects.  If you have time, please take what we have now for a spin and see if anything comes up that needs fixing before the release.

Ping @stevedlawrence @cgzones since I'm not sure how loud this new "Discussions" field is to get your attention.

**Reply by stevedlawrence** [reply]
Sounds good to me. FYI, I am "watching" this repo so I do get notifications, including the new discussion stuff. I'll try to find some time to play with it a bit.

**Reply by dburgener** [reply]
Good to know.  Wasn't sure whether watch included discussions.  Thanks!

**Comment by cgzones** [comment]
Sounds good; I'll do some final testing.

Btw. your new check `C-007 - Recommend use of self keyword instead of redundant type` has some findings in the Reference Policy.
Do you like to propose a patch?

```
xserver.te:         790: (C): Recommend use of self keyword instead of redundant type (C-007)
xserver.te:         792: (C): Recommend use of self keyword instead of redundant type (C-007)
kernel.te:          266: (C): Recommend use of self keyword instead of redundant type (C-007)
Found the following issue counts:
C-007: 3
```

**Reply by dburgener** [reply]
Thanks!

Thanks for the info on refpolicy.  I hadn't checked that.  Yes, I'll try to get a patch for that up today.

**Reply by dburgener** [reply]
refpolicy PR: https://github.com/SELinuxProject/refpolicy/pull/326

**Comment by gdesrosi** [comment]
@dburgener I just thought about the null check around res before freeing res->message in the free_check_result function that I was planning on getting in under a separate commit with #183 so given that is not going to make it in for the release perhaps we should get that in under its own PR?

**Reply by dburgener** [reply]
Great idea.  Yes please.

**Comment by dburgener** [comment]
Sorry about the slow progress here.  I think we're in a pretty good state.  Thinking about cutting the release this afternoon.  Any reason to hold off?
